include/ebpf_structs.h

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright (c) Microsoft Corporation
+ *  SPDX-License-Identifier: MIT
+ */
+
+// This file contains eBPF definitions common to eBPF programs, core execution engine
+// as well as eBPF API library.
```

Comment:
This header will be included by actual eBPF programs. So I dont want those programs to unnecessarily include types that are not needed for them. I like the proposal of renaming this to ebp_map.h. I will make that change.

Reply:
> I dont want those programs to unnecessarily include types that are not needed for them.

Why would you care?  Taken to an extreme, that would motivate every standard C API being in a separate header, which is clearly against the standard.   If you want to propose a guideline, please write up a justification and we can discuss.   Otherwise, ease of developer experience argues for minimizing the number of headers one must #inlcude.  And since unused types take up 0 bytes in the binary, I don't yet follow why one would want to explode the number of files.

Reply:
It makes documentation more complicated, e.g., by making the list of files at https://microsoft.github.io/ebpf-for-windows/files.html much longer, and directory listings, and such, which can confuse developers who want to know what to include or what to ignore.  I haven't yet heard any technical reason to explode the number of files.